### PR TITLE
Assorted fixes and cleanup

### PR DESCRIPTION
**Pull request by MHajoha**
This PR does some minor bits of cleanup and fixes the logging setup as follows:
* Hardcode `sys.argv` in `conftest.py`.
* Don't repeat the logging setup in `logging_fx` as that breaks things. Clear the log capture instead.

### CODE REVIEW

**Comment by theunkn0wn1** [comment]
The logging system was not entirely finished, and @shatteredbeam was in the process of correcting many of the points you address in this PR.

I was under the impression you would just be fixing the unix compliance issue and the sys,arvg thing for testing. 

If possible, please remove the logging-specific fixes so Shatt's work on the issue is not ill-spent

**Comment by MHajoha** [comment]
Shatt and I have been able to come to agree here and I've removed my changes to `setup_logging`, as I did admittedly get up in his business there, while keeping those to the logging test fixtures and logging tests themselves, so that our tests still run successfully.

**Comment by shatteredbeam** [comment]
MHajoa and I ave indeed come to an agreement. the test improvements are OK with me to remain.

**Comment by theunkn0wn1** [comment]
it appears all the merging I did today caused merge conflicts, sorry 'bout that :(

File: Modules/cli_manager.py [hunk 1]
```diff
@@ -16,40 +16,29 @@
...
-_parser.add_argument("--clean-log", help="Start with a clean log file.  "
-                                         "Deletes any existing log data.",
+_parser.add_argument("--clean-log",
+                     help="Start with a clean log file. Deletes any existing log data.",
                      action="store_true")
+
 # register optional argument for the config file
-_parser.add_argument("--config-file", "--config", help="Specify the configuration file to load, "
-                                                       "relative to config/",
+_parser.add_argument("--config-file", "--config", "-c",
```
Comment (theunkn0wn1) [inline Modules/cli_manager.py]: i don't agree with `-c` being a valid shorthand for this cli argument. imho single char arguments should be flags.

Comment (MHajoha) [reply Modules/cli_manager.py]: In terms of convention, there shouldn't really be a difference between regular options and 'long' options `--` in what they are used for. [The POSIX specification added options (`-o`), while the GNU specification added long options (`--option`) as a more easily understandable version of the same thing.](https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html)
Also, only one out of three of the flags we have has a short option, so we aren't really on that route anyways.
Interestingly, GNU wants long options with arguments to be referenced by `--option=arg`, but I don't think argparse supports this.

Comment (theunkn0wn1) [reply Modules/cli_manager.py]: grumbles

**Pull request merged**